### PR TITLE
ICS2: Generic Verification Methods

### PR DESCRIPTION
**Pull request by AdityaSripal**
Closes: #684

### CODE REVIEW

File: spec/core/ics-002-client-semantics/README.md [hunk 1]
```diff
@@ -429,26 +429,40 @@ type verifyNextSequenceRecv = (
   => boolean
 ```
 
-#### Optional Functions
+`verifyMembership` is a generic proof verification method which verifies a proof of the existence a value at a given `CommitmentPath` at the specified height.
```
Comment (mpoke) [inline spec/core/ics-002-client-semantics/README.md:432]:
```suggestion
`verifyMembership` is a generic proof verification method which verifies a proof of the existence of a value at a given `CommitmentPath` at the specified height.
```

File: spec/core/ics-002-client-semantics/README.md [hunk 2]
```diff
@@ -429,26 +429,40 @@ type verifyNextSequenceRecv = (
...
+  delayPeriodTime: uint64,
+  delayPeriodBlocks: uint64,
+  proof: CommitmentProof,
+  path: CommitmentPath,
+  value: bytes)
+  => boolean
+```
 
-`verifyPacketReceipt` verifies a proof of an incoming packet receipt at the specified port, specified channel, and specified sequence. It is needed only if the chain wishes to support new channel types other than ORDERED and UNORDERED.
+`verifyNonMembership` is a generic proof verification method which verifies a proof of absence at a given `CommitmentPath` at the specified height.
```
Comment (mpoke) [inline spec/core/ics-002-client-semantics/README.md:449]:
```suggestion
`verifyNonMembership` is a generic proof verification method which verifies a proof of absence of a given `CommitmentPath` at the specified height.
```

File: spec/core/ics-002-client-semantics/README.md [hunk 3]
```diff
@@ -429,26 +429,40 @@ type verifyNextSequenceRecv = (
   => boolean
 ```
 
-#### Optional Functions
+`verifyMembership` is a generic proof verification method which verifies a proof of the existence a value at a given `CommitmentPath` at the specified height.
+The caller is expected to construct the full `CommitmentPath` from a `CommitmentPrefix` and a standardized path. If the caller desires a particular delay period to be enforced,
```
Comment (mpoke) [inline spec/core/ics-002-client-semantics/README.md:433]:
```suggestion
The caller is expected to construct the full `CommitmentPath` from a `CommitmentPrefix` and a standardized path (as defined in [ICS 24](../ics-024-host-requirements/README.md#path-space)). If the caller desires a particular delay period to be enforced,
```

File: spec/core/ics-002-client-semantics/README.md [hunk 4]
```diff
@@ -429,26 +429,40 @@ type verifyNextSequenceRecv = (
...
+  delayPeriodBlocks: uint64,
+  proof: CommitmentProof,
+  path: CommitmentPath,
+  value: bytes)
+  => boolean
+```
 
-`verifyPacketReceipt` verifies a proof of an incoming packet receipt at the specified port, specified channel, and specified sequence. It is needed only if the chain wishes to support new channel types other than ORDERED and UNORDERED.
+`verifyNonMembership` is a generic proof verification method which verifies a proof of absence at a given `CommitmentPath` at the specified height.
+The caller is expected to construct the full `CommitmentPath` from a `CommitmentPrefix` and a standardized path. If the caller desires a particular delay period to be enforced,
```
Comment (mpoke) [inline spec/core/ics-002-client-semantics/README.md:450]:
```suggestion
The caller is expected to construct the full `CommitmentPath` from a `CommitmentPrefix` and a standardized path (as defined in [ICS 24](../ics-024-host-requirements/README.md#path-space)). If the caller desires a particular delay period to be enforced,
```

File: spec/core/ics-002-client-semantics/README.md (hunk 1 above)
Comment (mpoke) [inline spec/core/ics-002-client-semantics/README.md:432]: The names of these functions, i.e., `verifyMembership` and `verifyNonMembership`, are already used in ICS2 (see the Implementation strategies section). Also, both names are used in ICS23, but I guess that's okay.

**Review by mpoke — APPROVED** [review]
LGTM. Some minor comments bellow. The only issue that I see is that the two functions are also defined in Implementation strategies.

File: spec/core/ics-002-client-semantics/README.md (hunk 1 above)
Comment (AdityaSripal) [reply spec/core/ics-002-client-semantics/README.md]: Clarified that the example in this section is the ICS-23 methods

**Pull request merged**